### PR TITLE
feat(x/ecocredit): query balances by batch denom

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds a query for balances by batch denom

Closes: n/a

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/base/keeper/query_balances_test.go [hunk 1]
```diff
@@ -49,11 +48,11 @@ func TestQuery_Balances(t *testing.T) {
 	assert.Equal(t, 0, len(res.Balances))
 }
 
-func assertBalanceEqual(ctx context.Context, t *testing.T, k Keeper, received *types.BatchBalanceInfo, balance *api.BatchBalance) {
+func (s *baseSuite) assertBalanceEqual(received *types.BatchBalanceInfo, balance *api.BatchBalance) {
```
Comment (technicallyty) [inline x/ecocredit/base/keeper/query_balances_test.go:51]: changed this here as i realized it was a bit un-ergonomic before

**Review by ryanchristo — APPROVED** [review]
[tACK](https://hackmd.io/z3UC-UrWQKGbW6uzIu0GSg)

**Comment by ryanchristo** [comment]
changelog :pray:

**Comment by technicallyty** [comment]
> tACK

im getting `403 forbidden` on this link btw 🤔 do i need to be added to a hackmd group or smthn?

**Comment by ryanchristo** [comment]
> im getting 403 forbidden on this link btw thinking do i need to be added to a hackmd group or smthn?

I think hackmd changed the default permissions. You can view now.

**Pull request merged**